snuba/consumers/consumer_builder.py

```diff
@@ -237,11 +283,52 @@ def log_general_error(e: KafkaError) -> None:
 
         return StreamProcessor(consumer, self.raw_topic, strategy_factory, IMMEDIATE)
 
+    # We currently don't having slicing implemented for multistorage case
+    def build_multistorage_streaming_strategy_factory(
+        self,
+    ) -> ProcessingStrategyFactory[KafkaPayload]:
+
+        if self.commit_log_topic:
+            commit_log_config = CommitLogConfig(
+                self.commit_log_producer, self.commit_log_topic, self.group_id
+            )
+        else:
+            commit_log_config = None
+
+        dead_letter_policies = {
+            storage.get_table_writer()
+            .get_stream_loader()
+            .get_dead_letter_queue_policy_creator()
+            for storage in self.storages.values()
+        }
+
+        dead_letter_policy_creator = dead_letter_policies.pop()
+        if dead_letter_policies:
+            raise ValueError("only one dead letter policy is supported")
+
+        strategy_factory = MultistorageConsumerProcessingStrategyFactory(
+            [*self.storages.values()],
+            self.max_batch_size,
+            self.max_batch_time_ms / 1000.0,
+            processes=self.processes,
+            input_block_size=self.input_block_size,
+            output_block_size=self.output_block_size,
+            metrics=self.metrics,
+            dead_letter_policy_creator=dead_letter_policy_creator,
+            commit_log_config=commit_log_config,
+        )
+
+        return strategy_factory
+
```

Comment:
Where did this code come from? Can we remove the other implementation rather than keeping 2 around? Alternatively, you can call the existing implementation from here and bring it over as a follow up. I think you should avoid ever having 2 copies of the same code.

Reply:
Makes sense. I'll call the similar code that lives in multistorage_consumer here.